android/app/src/main/java/com/sample/react/TestAppReactNativeHost.kt

```diff
@@ -49,32 +52,34 @@ class TestAppReactNativeHost @Inject constructor(
     override fun createReactInstanceManager(): ReactInstanceManager {
         ReactMarker.logMarker(ReactMarkerConstants.BUILD_REACT_INSTANCE_MANAGER_START)
         val reactInstanceManager = ReactInstanceManager.builder()
-                .setApplication(application)
-                .setCurrentActivity(currentActivity)
-                .setJavaScriptExecutorFactory(javaScriptExecutorFactory)
-                .setBundleAssetName(bundleAssetName)
-                .setJSMainModulePath(jsMainModuleName)
-                .addPackages(packages)
-                .setUseDeveloperSupport(useDeveloperSupport && !useEmbeddedBundle)
```

Comment:
This is going to break switching between embedded and dev server.

Reply:
Maybe we can determine the initial value of `useEmbeddedValue` based on `bundleAssetName != null` instead?

Reply:
Right, I reproduced the crash. Initialising to `bundleAssetName != null` sounds like an idea.